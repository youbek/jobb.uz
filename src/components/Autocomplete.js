import React from 'react';
// DON'T DELETE THE FOLLOWING COMMENT!
/* global google */

class SearchBar extends React.Component {
  constructor(props) {
    super(props);
    this.autocompleteInput = React.createRef();
    this.autocomplete = null;
    this.handlePlaceChanged = this.handlePlaceChanged.bind(this);
  }

  componentDidMount() {
    this.autocomplete = new google.maps.places.Autocomplete(
      this.autocompleteInput.current,
      { types: ['(cities)'], componentRestrictions: { country: 'us' } },
    );

    this.autocomplete.addListener('place_changed', this.handlePlaceChanged);
  }

  handlePlaceChanged() {
    const place = this.autocomplete.getPlace();
    this.props.handlePlaceChange(place);
  }

  render() {
    return (
      <input
        className="form-control"
        ref={this.autocompleteInput}
        id="autocomplete"
        placeholder="Enter location"
        type="text"
        disabled={this.props.fetching}
      />
    );
  }
}

export default SearchBar;
