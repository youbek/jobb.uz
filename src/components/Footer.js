import React from "react";
import { Container, Col, Row } from "reactstrap";

function Footer() {
  return (
    <footer className="border-top mt-5">
      <Container>
        <Row>
          <Col className="footer">
            <p>
              This website is for educational purposes only. Built to showcase
              my skills in full stack development.{" "}
              <a href="https://yusufbek.com">Contact Me</a>
            </p>
            <p>Copyright Â© 2019 Yusufbek Alimatov. All rights reserved.</p>
          </Col>
        </Row>
      </Container>
    </footer>
  );
}

export default Footer;
