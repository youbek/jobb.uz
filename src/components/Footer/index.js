import React from "react";
import styled from "styled-components";

const FooterContainer = styled.div`
  background: #fbfbfb;
  border-top-left-radius: 3px;
  color: #494949;
  border: 1px solid #eee;
  padding: 0.5rem;
  font-size: 14px;
  position: fixed;
  bottom: 21px;
  bottom: 0;
  right: 0;
  border-right: none;
  border-bottom: none;
  cursor: pointer;
  z-index: 1;
`;

function Footer() {
  return (
    <FooterContainer>
      <div>by Â© Yusufbek Alimatov ðŸ‘‰</div>
    </FooterContainer>
  );
}

export default Footer;
