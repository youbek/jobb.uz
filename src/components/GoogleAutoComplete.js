import React from "react";
import PropTypes from "prop-types";

// DON'T DELETE THE FOLLOWING COMMENT!
/* global google */

class GoogleAutoComplete extends React.Component {
  constructor(props) {
    super(props);
    this.autocompleteInput = React.createRef();
    this.autocomplete = null;
    this.handlePlaceChanged = this.handlePlaceChanged.bind(this);
  }

  componentDidMount() {
    this.autocomplete = new google.maps.places.Autocomplete(
      this.autocompleteInput.current,
      {
        types: this.props.fullAddress ? ["address"] : ["(cities)"],
        componentRestrictions: { country: "us" },
      },
    );

    this.autocomplete.addListener("place_changed", this.handlePlaceChanged);
  }

  handlePlaceChanged() {
    const place = this.autocomplete.getPlace();
    this.props.handlePlaceChange(place);
  }

  render() {
    return (
      <input
        className="form-control"
        ref={this.autocompleteInput}
        id="autocomplete"
        placeholder="Enter location"
        type="text"
        disabled={this.props.fetching}
        autoComplete="false"
      />
    );
  }
}

export default GoogleAutoComplete;
