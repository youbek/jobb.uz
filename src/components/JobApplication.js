import React, { useState, useContext } from "react";

import { Button, Tooltip, Modal, ModalBody, ModalHeader } from "reactstrap";
import moment from "moment";
import { AuthContext } from "../context/AuthContext";

function JobApplication({ job }) {
  const { authenticatedUser } = useContext(AuthContext);
  // TOOLTIP STATE
  const [tooltipOpen, setTooltipOpen] = useState(false);
  const toggleTooltip = () => setTooltipOpen(!tooltipOpen);

  // CONTACT MODEL STATE
  const [modal, setModal] = useState(false);
  const toggleModal = () => setModal(!modal);

  // VALIDATING WHETHER JOB EXPIRED OR NOT
  const dateFrom = moment()
    .subtract(30, "d")
    .format("YYYY-MM-DD");
  const isJobActive = new Date(job.date) > new Date(dateFrom);

  if (authenticatedUser && authenticatedUser.isHr) {
    return (
      <div className="job-page-action-buttons mt-4">
        <Button id="userIsHr" className="mr-2" color="primary">
          Apply
        </Button>

        <Tooltip
          placement="top"
          target="userIsHr"
          isOpen={tooltipOpen}
          toggle={toggleTooltip}
        >
          Only job seekers can apply for jobs. You are signed in as an employer.
        </Tooltip>

        {job.contactPhone && (
          <Button color="warning" outline disabled>
            Show Contact
          </Button>
        )}
      </div>
    );
  }

  return (
    <div className="job-page-action-buttons mt-4">
      {isJobActive ? (
        <Button color="primary" className="mr-2">
          Apply
        </Button>
      ) : (
        <Button color="primary" className="mr-2" disabled>
          Expired Vacancy
        </Button>
      )}
      {job.contactPhone && (
        <React.Fragment>
          <Button onClick={toggleModal} outline color="secondary">
            Show Contact
          </Button>

          {
            // CONTACT PHONE MODAL
          }

          <Modal isOpen={modal} toggle={toggleModal}>
            <ModalHeader
              toggle={toggleModal}
            >{`${job.title} at ${job.companyName}`}</ModalHeader>
            <ModalBody>
              {`${job.author.firstName} ${job.author.lastName}`}
              <div>{job.contactPhone}</div>
            </ModalBody>
            <Button color="secondary" onClick={toggleModal}>
              Close
            </Button>
          </Modal>
        </React.Fragment>
      )}
    </div>
  );
}

export default JobApplication;
