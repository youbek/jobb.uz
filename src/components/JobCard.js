import React from "react";
import JobDescription from "./JobDescription";
import JobCardBadges from "./JobCardBadges";
import styled from "styled-components";
import Moment from "react-moment";
import "moment-timezone";
import { Link } from "react-router-dom";

const JobCardViewCompanyName = styled.div`
  font-weight: 700;
  font-size: 14px;
  margin-top: 15px;
`;
const JobCardViewFooter = styled.div`
  display: flex;
`;
const JobCardViewAddress = styled.div`
  font-size: 14px;
  color: #757575;
  margin-top: 4px;
`;

const JobCard = ({ job }) => {
  return (
    <div className="job-card">
      <div>
        <div className="job-card-date">
          <Moment locale="ru" fromNow>
            {job.date}
          </Moment>
        </div>
        <div className="job-card-job-title">{job.title}</div>
      </div>
      <JobDescription description={job.description} />
      <JobCardBadges noExperience={job.noExperience} partTime={job.partTime} />
      <JobCardViewCompanyName>{job.companyName}</JobCardViewCompanyName>
      <JobCardViewFooter>
        <JobCardViewAddress>{job.address}</JobCardViewAddress>
      </JobCardViewFooter>
      <Link className="job-card-link" to={`/vacancy/${job.hashId}`}></Link>
    </div>
  );
};

export default JobCard;
