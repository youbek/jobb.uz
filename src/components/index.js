// BUTTONS
export { default as Button } from "./Buttons/Button";
