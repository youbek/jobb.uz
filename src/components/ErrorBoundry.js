import React, { Component } from "react";
import { Container } from "components";

class ErrorBoundry extends Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    return { hasError: true };
  }

  componentDidCatch(error, errorInfo) {
    console.log(error, errorInfo);
  }

  render() {
    if (this.state.hasError) {
      return (
        <Container className="text-center mt-4">
          <h3>Something Went Wrong</h3>
          <a href="/">Return Home</a>
        </Container>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundry;
