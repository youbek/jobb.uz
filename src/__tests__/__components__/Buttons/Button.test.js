import React from "react";
import { render } from "@testing-library/react";
import { Button } from "components";
import "jest-styled-components";

test("Button component rendering", () => {
  const { container: primaryButton } = render(<Button>Hello World!</Button>);
  const { container: secondaryButton } = render(
    <Button secondary>Hello World!</Button>,
  );
  const { container: greyButton } = render(<Button grey>Hello World!</Button>);

  expect(primaryButton.firstChild).toMatchSnapshot();
  expect(secondaryButton.firstChild).toMatchSnapshot();
  expect(greyButton.firstChild).toMatchSnapshot();
});
