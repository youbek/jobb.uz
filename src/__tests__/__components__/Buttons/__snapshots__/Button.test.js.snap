// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component rendering 1`] = `
.c0 {
  display: inline-block;
  font-weight: 600;
  border: 0;
  color: #fff;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #F64F64;
  padding: 0.62rem 1.5rem;
  font-size: 14px;
  line-height: 1.5;
  border-radius: 0.25rem;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out,border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out,border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
}

.c0:hover {
  background-color: #db495b;
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<button
  class="c0"
>
  Hello World!
</button>
`;

exports[`Button component rendering 2`] = `
.c0 {
  display: inline-block;
  font-weight: 600;
  border: 0;
  color: #fff;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #4687FF;
  padding: 0.62rem 1.5rem;
  font-size: 14px;
  line-height: 1.5;
  border-radius: 0.25rem;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out,border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out,border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
}

.c0:hover {
  background-color: #3669c6;
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<button
  class="c0"
>
  Hello World!
</button>
`;

exports[`Button component rendering 3`] = `
.c0 {
  display: inline-block;
  font-weight: 600;
  border: 0;
  color: #5B5E64;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #fff;
  padding: 0.62rem 1.5rem;
  font-size: 14px;
  line-height: 1.5;
  border: 1px solid #CBD1D4;
  border-radius: 0.25rem;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out,border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out,border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
}

.c0:hover {
  background-color: #fff;
  color: #5B5E64;
  border: 1px solid #acb5ba;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<button
  class="c0"
>
  Hello World!
</button>
`;
